### PR TITLE
Add test for soft-reboot

### PR DESCRIPTION
**Pull request by sujinmkang**
### Description of PR

This is for adding a test for soft-reboot.
Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
To verify all dockers and services are running as expected after soft-reboot.
#### How did you do it?
Add the soft-reboot test to test_reboot.py
#### How did you verify/test it?
Run a pytest with following command after installing the soft-reboot on the DUT.
```
export PYTEST_SCRIPTS="platform_tests/test_reboot.py"
export SWITCH="<DUT_NAME>"
export TOPO="<TESTBED_TOPO>"
export JOB_NAME="test_reboot"
pytest ${PYTEST_SCRIPTS} --testbed=${TOPO} --inventory=../ansible/str --testbed_file=../ansible/testbed.csv --host-pattern=${SWITCH} --module-path=../ansible/library --show-capture=no -v -k "test_soft_reboot"
```

### CODE REVIEW

**Review by jleveque — APPROVED** [review]

**Review by yxieca — APPROVED** [review]

**Comment by yxieca** [comment]
@sujinmkang  can you update how you verified it?

**Comment by yxieca** [comment]
The reboot test so far has assumption that the reboot command is available on target device, so it didn't check if the command exists or not before executing it. Can you add a check? Otherwise, this test will fail on many branches until the cherry-pick is done.

**Comment by sujinmkang** [comment]
@yxieca can you review the change again?

File: tests/common/reboot.py [hunk 1]
```diff
@@ -44,6 +45,13 @@
         "cause": r"'reboot'|Non-Hardware \(reboot",
         "test_reboot_cause_only": False
     },
+    REBOOT_TYPE_SOFT: {
+        "command": "soft-reboot",
+        "timeout": 300,
+        "wait": 120,
```
Comment (vaibhavhd) [inline tests/common/reboot.py:51]: Is this 120s wait by design?

Comment (sujinmkang) [reply tests/common/reboot.py:51]: soft-reboot doesn't need any tight timing requirement. And it's followed cold reboot wait timing. Any concern?

Comment (vaibhavhd) [reply tests/common/reboot.py:51]: No concerns, I was just curious to know.

**Review by yxieca — APPROVED** [review]

**Comment by antonptashnik** [comment]
@sujinmkang Hi. Could you please point me to some soft-reboot mode explanation? I was not able to find any document unfortunately

**Pull request merged**